Feed Support for Listings

dragonstyle opened the pull request:
- Generate feeds for listing pages (optionally)
- Feeds support `partial` or `full-content`
- Feeds are page level, so will collect all the items from any listings on the page
- You can also generate category specific feeds like so:
```
listing:
  feed:
    categories:
      - category1
      - category2
```
- In addition, now listing pages will automatically render whenever a page they reference is rendered (or when a new file that matches their contents is rendered).

dragonstyle committed:
Baseline generation of feed need to listing page.

Baseline generation of feed from a listing page. Note that this still has issues with orphaning the feed files.

dragonstyle committed:
Add support for rendering category feeds as well

dragonstyle committed:
Allow HTML Post Processors to return resources or supporting files

(Resources are copied, supporting files are moved)

dragonstyle committed:
Add core RSS feed to document

dragonstyle committed:
Add category feed link tags

dragonstyle committed:
Rough in completing full feeds

dragonstyle committed:
Implement full and partial feed contents

First stage the feeds with placeholders, then read the output contents to replace the title, description

dragonstyle committed:
Fix path, strip code copy button

dragonstyle committed:
Apply syntax highlighting styles to code in the feed

dragonstyle committed:
Add Webtex math support

dragonstyle committed:
Improve image path handling

dragonstyle committed:
fix atom link

dragonstyle committed:
Remove internal links keeping their contents

dragonstyle committed:
cleanup feedlink

dragonstyle committed:
Properly resolve feed item images and properties

dragonstyle committed:
Improve Feed Metadata

- pubDate is set by the most recent item (or now)
- image for the feed itself is now resolve properly

dragonstyle committed:
Require a title to render a feed

dragonstyle committed:
Scale feed images appropriately

dragonstyle committed:
Generate even empty feeds

dragonstyle committed:
default to full content feeds

dragonstyle committed:
update category image props

dragonstyle committed:
Introduce project type ability to supplement render files

dragonstyle committed:
Don't look for staged feeds if the site-url isn't set

No need to error b/c we always run this step (and the previous step won't emit a staged file if the listing can't be created).

dragonstyle committed:
Extend supplemental files

- mark supplemental files so they can be treated specially
- don't output supplemental files
- pass project and incremental flag to supplemental files hook

dragonstyle committed:
Listings can now add supplemental files to rendereng

Listings will keep a cache of listing files and the globs that they use. When a file is rendered, we will check the globs to see whether they match the rendered, and if so, we will also render that listing page.

dragonstyle committed:
Cleanup staged feeds if not completed successfully for any reason

jjallaire commented on src/project/types/website/listing/website-listing-shared.ts:
```diff
@@ -87,9 +96,20 @@ export type CategoryStyle =
+export interface ListingFeedOptions {
+  [kTitle]?: string;
+  [kItems]?: number;
+  [kType]: "partial" | "full";
```
Perhaps this should be `summary` and `full` ?

jjallaire commented on src/project/types/website/website-config.ts:
```diff
@@ -103,9 +104,11 @@ export interface OpenGraphConfig {
 
 type WebsiteConfigKey =
   | "title"
+  | "image"
```
Be sure to add `image` and `site-description` to the website schema (or just note that it should be done when creating the documentation)

dragonstyle replied:
Will do as a part of listing schema/doc update.

dragonstyle committed:
partial=>summary

dragonstyle committed:
site-description -> description

dragonstyle committed:
Delist category feeds

- category feeds have the same name as the main feed
- category feeds aren't included as link tags in the document

dragonstyle committed:
Restore language support
